detanalysis/semiautocuts.py

```diff
@@ -1825,31 +1895,125 @@ def combine_cuts(self, sat_pass_threshold=None, cut_name=None):
             
         cut_name : string, optional
             If not none, this is the name of the cut created.
+            
+        lgc_diagnostics : bool, optional
+            If True, prints out diagnostic statements
         
         """
         
         self.cut_name = cut_name
         
         cuts_all_arr = np.ones(len(self.df), dtype = 'bool')
         
+        if cuts_to_combine is None:
+            cuts_to_combine = self.cuts_list
+            
+        if lgc_diagnostics:
+            print("Combining cuts: " + str(cuts_to_combine))
+            
         i = 0
-        while i < len(self.cuts_list):
-            current_cut_arr = self.df[self.cuts_list[i]].values
+        while i < len(cuts_to_combine):
```

Comment:
Just checking, but cuts_to_combine and self.cuts_list fundamentally contain different things right? Or at least they may? If cuts_to_combine is none would you not be combining things you didn't originally intend to?

Reply:
I'll change this, I think you're right it can be better